### PR TITLE
Add new typography method in Caption component

### LINKED ISSUE #345: Implement new typography in Caption component
Blocked by #342 

**Is your feature request related to a problem? Please describe.**
World Service has a UX requirement that GEL Types (Canon, Long Primer, Double Pica, etc) should allow for mutable values across services. These values were grouped into 11 different scripts: `arabic`, `arabic-pashto`, `bengali`, `burmese`, `devanagari-and-gurmukhi`, `latin-and-cyrillic`, `latin-with-diacritics`, `no-ascenders-or-descenders`, `sinhalese`, `tamil`, and `thai`.

The `latin-and-cyrillic` script contains the default values GEL for all the Types. The previous constants (GEL_CANON, GEL_LONG_PRIMER, GEL_DOUBLE_PICA, etc) are being deprecated in favour of script-specific methods such as `getCanon()`, `getLongPrimer()` and so on. These methods take the current component's script as an argument.

This issue is part of a wider work to update all existing components to use the new methods, eventually removing the old constants from the codebase.

**Describe the solution you'd like**
* Allow the component to pass the script name and children as a prop.
* Update the component's propTypes and defaultProps to include a prop of `script` that defaults to `latin`
* Import the appropriate getTypeSizes() method for the component's Type Styles (getLongPrimer)
* Implement the method above, replacing any references to the deprecated GEL Type constant (GEL_LONG_PRIMER) in the styled-component CSS string 
* Remove existing imports of the previous GEL Type constant (GEL_LONG_PRIMER)

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

- [ ] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #345

**Overall change:** 
Add support to allow the `Caption` component to use Typography Type Sizes for different scripts

**Code changes:**

- Allow the component to pass the script name a prop
- Add `propTypes` to require a `script` prop
- Remove previous `GEL_LONG_PRIMER` import
- Import new `getLongPrimer` method which makes use of the new `getTypeSizes()` method to get the appropriate GEL Type Sizes for the component.
- Import `latin.js` script with the default typography GEL Types Sizes in the stories and unit test
---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-caption/src/index.jsx [hunk 1]
```diff
@@ -10,20 +12,21 @@ import {
...
 
 // Defined separately since in future will need to apply
 // only when the script supports italic text
 const FS_ITALIC = css`
   font-style: italic;
 `;
 
-const Caption = styled.figcaption`
-  ${GEL_LONG_PRIMER};
+const StyledElement = styled.figcaption`
```
Comment (pjlee11) [inline packages/components/psammead-caption/src/index.jsx]:
```suggestion
const StyledCaption = styled.figcaption`
```

File: packages/components/psammead-caption/src/index.jsx [hunk 2]
```diff
@@ -38,4 +41,19 @@ const Caption = styled.figcaption`
   }
 `;
 
+const Caption = ({ children, script }) => {
+  const GEL_LONG_PRIMER = getLongPrimer(script);
+
+  return <StyledElement typography={GEL_LONG_PRIMER}>{children}</StyledElement>;
```
Comment (pjlee11) [inline packages/components/psammead-caption/src/index.jsx]:
```suggestion
  return <StyledCaption typography={GEL_LONG_PRIMER}>{children}</StyledCaption>;
```

Comment (DenisHdz) [reply packages/components/psammead-caption/src/index.jsx]: @pjlee11 Shall we apply the same naming `<StyledComponentName/>` to the rest of components?

File: packages/components/psammead-caption/src/index.jsx [hunk 3]
```diff
@@ -10,20 +12,21 @@ import {
   GEL_GROUP_4_SCREEN_WIDTH_MIN,
 } from '@bbc/gel-foundations/breakpoints';
 import {
-  GEL_LONG_PRIMER,
+  getLongPrimer,
   GEL_FF_REITH_SANS,
 } from '@bbc/gel-foundations/typography';
 
 import { C_CLOUD_DARK } from '@bbc/psammead-styles/colours';
+import { latin } from '@bbc/gel-foundations/scripts';
```
Comment (pjlee11) [inline packages/components/psammead-caption/src/index.jsx]: I'm interested in figuring out if this imports `latin` even when the default isn't used. If it does always import it we should move the responsibility of passing the script to the container and make `script` a required prop.

Comment (DenisHdz) [reply packages/components/psammead-caption/src/index.jsx]: Yep, you are right 👍🏻

**Comment by pjlee11** [comment]
Good work so far!

There are some conflicts. Also could you add a snapshot test for one of non-latin scripts please

File: packages/components/psammead-caption/CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
-| 0.3.4   | [PR#407](https://github.com/bbc/psammead/pull/407) Organise dependencies properly |
```
Comment (andrew-nowak) [inline packages/components/psammead-caption/CHANGELOG.md]: Don't want to delete from the changelog

File: packages/components/psammead-caption/package.json [hunk 5]
```diff
@@ -17,8 +17,9 @@
   },
   "homepage": "https://github.com/bbc/psammead/blob/latest/packages/components/psammead-caption/README.md",
   "dependencies": {
-    "@bbc/gel-foundations": "^0.2.5",
-    "@bbc/psammead-styles": "^0.3.2"
+    "@bbc/gel-foundations": "^1.0.0",
+    "@bbc/psammead-styles": "^0.3.2",
+    "styled-components": "^4.1.2"
```
Comment (andrew-nowak) [inline packages/components/psammead-caption/package.json]: Looks like a merge issue, I think styled components are in devDependencies only now

**Review by pjlee11 — DISMISSED** [review]

**Review by pjlee11 — APPROVED** [review]
Nice work 👍

**Review by dr3 — APPROVED** [review]

**Pull request merged**